File: master-package/bonus-webinars-2018/introduction-to-spring-cloud-contract/rest-service/src/main/java/com/testwithspring/master/webinar/Task.java
package com.testwithspring.master.webinar;

import org.apache.commons.lang3.builder.ToStringBuilder;

/**
 * Contains the information of a single tasdk.
 */
public class Task {

    private final Long id;
    private final String title;

    public Task(Long id, String title) {
        this.id = id;
        this.title = title;
    }

    public Long getId() {
        return id;
    }

    public String getTitle() {
        return title;
    }

    @Override
    public String toString() {
        return new ToStringBuilder(this)
                .append("id", this.id)
                .append("title", this.title)
                .toString();
    }
}

File: master-package/bonus-webinars-2019/introduction-to-mockk/src/main/kotlin/com/testwithspring/master/kotlin/Task.kt
package com.testwithspring.master.kotlin

/**
 * Contains the user id of the user who created a task.
 */
data class Creator(val id: Long)

/**
 * Defines the resolutions of a task.
 */
enum class TaskResolution {
    DUPLICATE,
    DONE,
    WONT_DO
}

/**
 * Defines the statuses of a task.
 */
enum class TaskStatus {
    CLOSED,
    IN_PROGRESS,
    OPEN
}

/**
 * Contains the information of a single task.
 */
data class Task(val id: Long?,
                val creator: Creator,
                val title: String,
                val status: TaskStatus,
                val resolution: TaskResolution?)